lib/hcaptcha_verifier.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module HcaptchaVerifier
+  class << self
+    VERIFY_ENDPOINT = "https://hcaptcha.com/siteverify"
+    SIGN_IN_CAPTCHA_THRESHOLD = 4
+    SIGN_UP_CAPTCHA_THRESHOLD = 2
+
+    def should_verify_sign_in?(user_display_id)
+      key = Rack::Attack::LOGIN_THROTTLE_PER_USER_KEY
+      period = Rack::Attack::LOGIN_LIMIT_PERIOD
+
+      keys = keys(key, period, user_display_id)
+      should_verify(keys, SIGN_IN_CAPTCHA_THRESHOLD)
+    end
+
+    def should_verify_sign_up?(ip)
+      key = Rack::Attack::SIGN_UP_THROTTLE_PER_IP_KEY
+      period = Rack::Attack::SIGN_UP_LIMIT_PERIOD
+
+      keys = keys(key, period, ip)
+      should_verify(keys, SIGN_UP_CAPTCHA_THRESHOLD)
+    end
+
+    def call(client_response, remote_ip)
+      return false unless client_response
+
+      payload = {
+        response: client_response,
+        remoteip: remote_ip,
+        secret: ENV.fetch("HCAPTCHA_SECRET", "0x0000000000000000000000000000000000000000"),
```

Comment:
these env var values should go into `config/deploy/*/secrets.ejson`, and then they should be added to the unicorn k8s container definition as env vars

Reply:
I don't have the actual values to use in deployed environments. I used only demo accounts for my development but this may require someone with creds for hcaptcha and Fastly to grab.